lib/options.nix

```diff
@@ -6,6 +6,34 @@
 with lib;
 with nixvimUtils;
 rec {
+  # Render a plugin default string
+  pluginDefaultText =
+    {
+      # plugin default, used if `defaultText` is missing
+      default ? null,
+      # pre-formatted plugin default string
+      #
+      # Assume a string `default` is already formatted as intended,
+      # TODO: remove this behavior so we can quote strings properly
+      # historically strings were the only type accepted by mkDesc.
+      defaultText ? if isString default then default else generators.toPretty { } default,
+      # nix option default value, used if `nixDefaultText` is missing
+      nixDefault ? null,
+      # pre-formatted nix option default string
+      nixDefaultText ? generators.toPretty { } nixDefault,
```

Comment:
I think these names would be clearer if we used `pdefault` or `pluginDefault`:

- `default`: the actual value used wen option is missing (e.g. `null`)
- `defaultText`: any value, or literal expression, to display instead of `default`
- `pluginDefault`: any value, or literal expression, to display as **"plugin default"**

Reply:
Partially resolved: `defaultNullOpts` still uses `default`, but `convertArgs` translates that into `pluginDefaults`.

Eventually, I think we should migrate `defaultNullOpts`'s `default` -> `pluginDefault` too.

Reply:
> I think these names would be clearer if we used `pdefault` or `pluginDefault`:
> 
>     * `default`: the actual value used wen option is missing (e.g. `null`)
> 
>     * `defaultText`: any value, or literal expression, to display instead of `default`
> 
>     * `pluginDefault`: any value, or literal expression, to display as **"plugin default"**

I agree with this naming.



> I think trying to use `default` to mean "plugin default" in the `defaultNullOpts` context has caused unnecessary confusion.

Yes, that's not optimal.